File: LeetCode/[E] - 1071. Greatest Common Divisor of Strings/solution.go
package main

import "strings"

func gcdOfStrings(str1 string, str2 string) string {
	var mnLn int
	if len(str1) > len(str2) {
		mnLn = len(str2)
	} else {
		mnLn = len(str1)
	}
	for i := mnLn; i > 0; i-- {
		if len(str1)%i == 0 && len(str2)%i == 0 {
			cand := str1[:i]
			div1 := strings.Split(str1, cand)
			div2 := strings.Split(str2, cand)
			if strings.Join(div1, "") == "" && strings.Join(div2, "") == "" {
				return cand
			}
		}
	}
	return ""
}

File: LeetCode/[E] - 1071. Greatest Common Divisor of Strings/solution.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        len_str1, len_str2 = len(str1), len(str2)
        for i in range(min(len_str1, len_str2), -1, -1):
            if len_str1 % (i + 1) == 0 and len_str2 % (i + 1) == 0:
                cand = str1[: i + 1]
                div1 = str1.split(cand)
                div2 = str2.split(cand)
                if "".join(div1) == "".join(div2) == "":
                    return cand
        return ""
